app/src/components/ActionsPanel.js

```diff
@@ -1,137 +1,157 @@
+import React, { useCallback, useMemo, useState } from 'react'
+import {
+  Button,
+  GU,
+  Info,
+  Field,
+  SidePanel,
+  TextInput,
+  useSidePanelFocusOnReady,
+  useTheme,
+} from '@aragon/ui'
+import wrap from '../assets/wrap.svg'
+import { fromDecimals, toDecimals } from '../utils'
 
-const initialState = {
-  amount: '',
-}
+// Any more and the number input field starts to put numbers in scientific notation
+const MAX_INPUT_DECIMAL_BASE = 6
 
 const WrapTokensPanel = React.memo(
-  ({
-    panelState,
-    onAction,
-    action,
-    info,
-    erc20TokenSymbol,
-    wrappedTokenSymbol,
-  }) => {
+  ({ action, info, onAction, outsideToken, panelState, wrappedToken }) => {
     return (
       <SidePanel
         title={action + ' tokens'}
         opened={panelState.visible}
         onClose={panelState.requestClose}
-        onTransitionEnd={panelState.onTransitionEnd}
       >
         <WrapTokensPanelContent
-          onAction={onAction}
           action={action}
           info={info}
-          erc20TokenSymbol={erc20TokenSymbol}
-          wrappedTokenSymbol={wrappedTokenSymbol}
-          panelOpened={panelState.didOpen}
+          onAction={onAction}
+          outsideToken={outsideToken}
+          wrappedToken={wrappedToken}
         />
       </SidePanel>
     )
   }
 )
 
-class WrapTokensPanelContent extends React.PureComponent {
-  static defaultProps = {
-    onAction: () => {},
-  }
-  state = {
-    ...initialState,
-  }
-  componentWillReceiveProps({ panelOpened }) {
-    if (panelOpened && !this.props.panelOpened) {
-      // setTimeout is needed as a small hack to wait until the input's on
-      // screen until we call focus
-      this.amountInput && setTimeout(() => this.amountInput.focus(), 0)
-    } else if (!panelOpened && this.props.panelOpened) {
-      // Finished closing the panel, so reset its state
-      this.setState({ ...initialState })
-    }
-  }
-  handleAmountChange = event => {
-    this.setState({ amount: event.target.value })
-  }
+function WrapTokensPanelContent({
+  action,
+  info,
+  onAction,
+  outsideToken,
+  wrappedToken,
+}) {
+  const theme = useTheme()
 
-  handleSubmit = event => {
-    event.preventDefault()
-    this.props.onAction(this.state.amount)
-  }
-  render() {
-    const { amount } = this.state
+  const [amount, setAmount] = useState('')
+  const tokenInputRef = useSidePanelFocusOnReady()
 
-    return (
-      <div>
-        <form
-          css={`
-            margin-top: ${3 * GU}px;
-          `}
-          onSubmit={this.handleSubmit}
-        >
+  const tokenData = useMemo(
+    () => (action === 'Wrap' ? outsideToken : wrappedToken),
+    [action, outsideToken, wrappedToken]
+  )
+  const handleAmountChange = useCallback(event => {
+    setAmount(event.target.value)
+  }, [])
+  const handleSubmit = useCallback(
+    event => {
+      event.preventDefault()
+
+      onAction(toDecimals(amount.trim(), tokenData.numDecimals))
```

Comment:
This ensures we handle token decimals correctly.

For example, if a token has 18 decimals, we need to transform a user's input of `1` to `1000000000000000000`.